src/runtime-rs/crates/hypervisor/src/qemu/qmp.rs

```diff
@@ -489,6 +491,119 @@ impl Qmp {
 
         Err(anyhow!("no target device found"))
     }
+
+    /// hotplug block device:
+    /// {
+    ///     "execute": "blockdev-add",
+    ///     "arguments": {
+    ///         "node-name": "drive-0",
+    ///         "file": {"driver": "file", "filename": "/path/to/block"},
+    ///         "cache": {"direct": true},
+    ///         "read-only": false
+    ///     }
+    /// }
+    ///
+    /// {
+    ///     "execute": "device_add",
+    ///     "arguments": {
+    ///         "id": "drive-0",
+    ///         "driver": "virtio-blk-pci",
+    ///         "drive": "drive-0",
+    ///         "addr":"0x0",
+    ///         "bus": "pcie.1"
+    ///     }
+    /// }
+    pub fn hotplug_block_device(
+        &mut self,
+        block_driver: &str,
+        block_device: &BlockDevice,
```

Comment:
Frankly, just like in #10362, I'd like to avoid making the qmp module depend on types from parent modules if at all possible.  Could we consider making this function take the device id, path and the flags explicitly and return the `pci_path`?

Reply:
Fine for me about your idea of this. Though I have ever though as your said, but I found so many arguments listed in the method, then I pass the `block_device: &BlockDevice`. I think we can reach the deal at this point just passing specific parameters to this function.

Reply:
Thanks @Apokleos !  It might be a small thing but I feel it might come handy at some point that `qmp.rs` only depends on primitive/simple types and inner coupling is reduced.